contracts/DeprecateERC20.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports ============
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/*
+DeprecateERC20
+by Anna Carroll
+*/
+contract DeprecateERC20 {
+    // ============ Immutables ============
+
+    IERC20 public immutable oldToken;
+    uint256 public immutable exchangeRate;
+
+    // ============  Public Storage ============
+
+    IERC20 public newToken;
+    // amount of oldToken migrated; target = 25k (all crowdfund tokens)
+    uint256 public totalMigrated;
+
+    // ============  Events ============
+
+    event Migrated(address indexed owner, uint256 oldTokenAmount);
+
+    // ======== Constructor =========
+
+    constructor(address _oldToken, uint256 _exchangeRate) {
```

Comment:
wdyt about adding an emergency function for the mutlisig to be able to withdraw non-migrated tokens?

worried about us messing up the exchangeRate or something and not being able to get tokens out.  do we have a step-by-step plan for how we're going to test this on mainnet once we deploy? (for example: would prefer to only deposit a small amount in here to ensure exchangeRate is as expected before depositing whole amount)

Reply:
i think we can probably feel pretty safe that if we check the exchange rate after we deploy that it will be fine. however, i am not opposed to doing a test with my own small number of crowdfund tokens before we send the rest of the new tokens to the contract after we deploy, if it makes you more comfortable.

it feels kind of unnecessary to me to bake in an escape hatch for this, as long as we double check all of the numbers